storage/innobase/srv/srv0srv.cc

```diff
@@ -3495,3 +3505,166 @@ srv_is_undo_tablespace(
 	       && space_id < (srv_undo_space_id_start
 			      + srv_undo_tablespaces_open));
 }
+
+/** Enable the undo log encryption if it is set.
+It will try to enable the undo log encryption and write the metadata to
+undo log file header, if innodb_undo_log_encrypt is ON. */
+static
+void
+srv_enable_undo_encryption_if_set()
+{
+	fil_space_t*	space;
+	const char*	cant_set_undo_tablespace = "Can't set undo tablespace";
+	const char*	to_be_encrypted = " to be encrypted";
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	/* Check if encryption for undo log is enabled or not. If it's
+	   enabled, we will store the encryption metadata to the space header
+	   and start to encrypt the undo log block from now on. */
+	if (srv_undo_log_encrypt) {
+		if (srv_undo_tablespaces == 0) {
+			srv_undo_log_encrypt = false;
+			ib::error() << cant_set_undo_tablespace << "s"
```

Comment:
Constructing messages like that is very hostile to i18n. Not sure something we should address though